### ISSUE TITLE
Version 5.3.1 new Icons is breaking change for some

### ISSUE THREAD

**Issue by lclarkg18** [callstack/react-native-paper#3744]
### Current behaviour
On updating from version 5.3.1 if one is already using custom icons in other places in their app, upgrading to version 5.3.1 will cause fail to render icons.

I'm not exactly sure what's going on behind the scenes but my impression is that my own RN vector icons taken from ```createIconSetFromIcoMoon(iconConfig)``` is overriding the packages icons so the warning I get when using the searchbar is:
```
Failed prop type: Invalid prop 'name' of value 'close' supplied to Icon, expected one of - my long list of icons -.
```
Same for 'magnify'

### Expected behaviour
Having custom icons should not take precedence if no icon is set. I should also say that I have used onIconPress but I don't want to change the search icon necessarily.

### How to reproduce?
Perhaps use a custom icon set and check if it overwrites RNPaper icons? Let me know if a repo is essential for this; it's a lot of work or if the origin of the issue is quick to be spotted.

### What have you tried so far?
Can be fixed by: changing icons in the source code or adding icons of your own with the same name

### Your Environment

| software                      | version
| --------------------- | -------
| ios                                | x
| android                        | x
| react-native                | 0.71.4
| react-native-paper    | 5.3.1
| node                            | x.x.x
| npm or yarn                | x.x.x
| expo sdk                     | x.x.x

**Comment by lclarkg18** [comment]
I also had trouble downgrading for some reason, but that's probably just me. I've just added the two icons to my icon set for now

**Comment by lukewalczak** [comment]
Hey @lclarkg18, could you please prepare the expo snack presenting your issue?

**Comment by lclarkg18** [comment]
https://snack.expo.dev/@lclarkg18/searcher-and-icomoon-in-rnpaper-5-3-1

**Comment by lukewalczak** [comment]
Hey @lclarkg18, you are right, I overlooked the fact that PR I previously created and merged introduced a breaking change, that's why some components were not working as expected. 

I [reverted](https://github.com/callstack/react-native-paper/pull/3745) it and released `5.4.0`.

Thanks for the feedback and sorry for the confusion.

**Comment by lclarkg18** [comment]
Thanks for the quick turnaround Luke! I really appreciate your work
